ingest/Maybe.py

```diff
@@ -0,0 +1,55 @@
+__author__ = 'Stephan'
+
+from itertools import chain
+
+class Maybe:
+    def __init__(self, v=None):
+        self.value = v
+
+    @staticmethod
+    def nothing():
+        return Maybe()
+
+    @staticmethod
+    def of(t):
```

Comment:
Since we are separating this class into its own file, I will add comments to the new Maybe.py class after this pull request is merged.